# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/basetypes/Accessible.h
# Commit: 48dc94da9565
# Full Hash: 48dc94da9565d7f804092209a27821d729ccde60
# Author: James Teh <jteh@mozilla.com>
# Date: 2022-04-02 09:44:13
# Regressor Bug: 1735970
# File Overlap Count: 1
# Description:
#   Bug 1735970 part 3: Add TableAccessibleBase and TableCellAccessibleBase. r=morgan
#   
#   These are needed to support RemoteAccessible tables.
#   Stuff specific to LocalAccessible is still in TableAccessible and TableCellAccessible, which now inherit from the new Base classes.
#   
# ==============================================================================

diff -r 477c59860bba -r 48dc94da9565 accessible/basetypes/Accessible.h
--- a/accessible/basetypes/Accessible.h	Fri Apr 01 09:49:56 2022 +0000
+++ b/accessible/basetypes/Accessible.h	Fri Apr 01 09:49:57 2022 +0000
@@ -24,6 +24,8 @@
 class HyperTextAccessibleBase;
 class LocalAccessible;
 class RemoteAccessible;
+class TableAccessibleBase;
+class TableCellAccessibleBase;
 
 /**
  * Name type flags.
@@ -432,6 +434,9 @@
 
   virtual HyperTextAccessibleBase* AsHyperTextBase() { return nullptr; }
 
+  virtual TableAccessibleBase* AsTableBase() { return nullptr; }
+  virtual TableCellAccessibleBase* AsTableCellBase() { return nullptr; }
+
   /**
    * Return the localized string for the given key.
    */